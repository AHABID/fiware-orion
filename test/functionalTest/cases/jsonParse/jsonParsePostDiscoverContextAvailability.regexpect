+++++ Starting +++++++++++++++
HTTP/1.1 200 OK
Content-Length: 75
Content-Type: application/json
Date: REGEX(.*)

{
    "duration": "PT1M",
    "registrationId": "REGEX([0-9a-f]{24})"
}
+++++ Registration 1 Done +++++++++++++++
HTTP/1.1 200 OK
Content-Length: 607
Content-Type: application/json
Date: REGEX(.*)

{
    "contextRegistrationResponses": [
        {
            "contextRegistration": {
                "attributes": [
                    {
                        "isDomain": "false", 
                        "name": "temperature", 
                        "type": "degree"
                    }
                ], 
                "entities": [
                    {
                        "id": "ConferenceRoom", 
                        "isPattern": "false", 
                        "type": "Room"
                    }, 
                    {
                        "id": "OfficeRoom", 
                        "isPattern": "false", 
                        "type": "Room"
                    }
                ], 
                "providingApplication": "http://192.168.100.1:70/application"
            }
        }
    ]
}
+++++ Discovery 1 Done +++++++++++++++
+++++ Register 2: two attributes in vector +++++++++++++++
HTTP/1.1 200 OK
Content-Length: 75
Content-Type: application/json
Date: REGEX(.*)

{
    "duration": "PT1M",
    "registrationId": "REGEX([0-9a-f]{24})"
}
+++++ Registration 2 Done +++++++++++++++
HTTP/1.1 200 OK
Content-Length: 599
Content-Type: application/json
Date: REGEX(.*)

{
    "contextRegistrationResponses": [
        {
            "contextRegistration": {
                "attributes": [
                    {
                        "isDomain": "false", 
                        "name": "temperature", 
                        "type": "degree"
                    }, 
                    {
                        "isDomain": "false", 
                        "name": "pressure", 
                        "type": "bar"
                    }
                ], 
                "entities": [
                    {
                        "id": "BathRoom", 
                        "isPattern": "false", 
                        "type": "Room"
                    }
                ], 
                "providingApplication": "http://192.168.100.1:70/application"
            }
        }
    ]
}
+++++ Discovery 2 Done +++++++++++++++
