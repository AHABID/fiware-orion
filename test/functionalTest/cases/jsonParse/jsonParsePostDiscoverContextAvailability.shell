echo "+++++ Starting +++++++++++++++"
url="/NGSI9/registerContext"
payload='{
  "contextRegistrations": [
    {
      "entities": [
        {
          "type": "Room",
          "isPattern": "false",
          "id": "ConferenceRoom"
        },
        {
          "type": "Room",
          "isPattern": "false",
          "id": "OfficeRoom"
        }
      ],
      "attributes": [
        {
          "name": "temperature",
          "type": "degree",
          "isDomain": "false",
          "metadatas": [
            {
              "name": "ID",
              "type": "string",
              "value": "1110"
            }
          ]
        }
      ],
      "metadatas": [
        {
          "name": "ID",
          "type": "string",
          "value": "2212"
        }
      ],
      "providingApplication": "http://192.168.100.1:70/application"
    }
  ],
  "duration": "PT1M"
}'
curlJson ${url} "${payload}"

echo "+++++ Registration 1 Done +++++++++++++++"

url="/NGSI9/discoverContextAvailability"
payload='{
  "entities": [
    {
      "type": "Room",
      "isPattern": "false",
      "id": "ConferenceRoom"
    },
    {
      "type": "Room",
      "isPattern": "false",
      "id": "OfficeRoom"
    }
  ],
  "attributes": [
    "temperature",
    "pressure"
  ],
  "restriction": {
    "attributeExpression" : "Attribute Expression",
    "scopes": [
      {
        "type" : "st1",
        "value" : "sv1"
      },
      {
        "type" : "st2",
        "value" : "sv2"
      }
    ]
  }
}'
curlJson ${url} "${payload}"

echo "+++++ Discovery 1 Done +++++++++++++++"
echo "+++++ Register 2: two attributes in vector +++++++++++++++"

url="/NGSI9/registerContext"
payload='{
  "contextRegistrations": [
    {
      "entities": [
        {
          "type": "Room",
          "isPattern": "false",
          "id": "BathRoom"
        }
      ],
      "attributes": [
        {
          "name": "temperature",
          "type": "degree",
          "isDomain": "false",
          "metadatas": [
            {
              "name": "ID",
              "type": "string",
              "value": "1110"
            }
          ]
        },
        {
          "name": "pressure",
          "type": "bar",
          "isDomain": "false",
          "metadatas": [
            {
              "name": "ID",
              "type": "string",
              "value": "1110"
            }
          ]
        }
      ],
      "metadatas": [
        {
          "name": "ID",
          "type": "string",
          "value": "2212"
        }
      ],
      "providingApplication": "http://192.168.100.1:70/application"
    }
  ],
  "duration": "PT1M"
}'
curlJson ${url} "${payload}"

echo "+++++ Registration 2 Done +++++++++++++++"

url="/NGSI9/discoverContextAvailability"
payload='{
  "entities": [
    {
      "type": "Room",
      "isPattern": "false",
      "id": "BathRoom"
    }
  ],
  "attributes": [
    "temperature",
    "pressure"
  ],
  "restriction": {
    "attributeExpression" : "Attribute Expression",
    "scopes": [
      {
        "type" : "st1",
        "value" : "sv1"
      },
      {
        "type" : "st2",
        "value" : "sv2"
      }
    ]
  }
}'
curlJson ${url} "${payload}"

echo "+++++ Discovery 2 Done +++++++++++++++"
