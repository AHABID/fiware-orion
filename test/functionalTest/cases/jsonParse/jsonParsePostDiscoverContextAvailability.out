+++++ Starting +++++++++++++++
HTTP/1.1 200 OK
Content-Length: 75
Content-Type: application/json
Date: Mon, 02 Jun 2014 08:17:10 GMT

{
    "duration": "PT1M", 
    "registrationId": "538c3306c97e3502d1030c31"
}
+++++ Registration 1 Done +++++++++++++++
HTTP/1.1 200 OK
Content-Length: 607
Content-Type: application/json
Date: Mon, 02 Jun 2014 08:17:11 GMT

{
    "contextRegistrationResponses": [
        {
            "contextRegistration": {
                "attributes": [
                    {
                        "isDomain": "false", 
                        "name": "temperature", 
                        "type": "degree"
                    }
                ], 
                "entities": [
                    {
                        "id": "ConferenceRoom", 
                        "isPattern": "false", 
                        "type": "Room"
                    }, 
                    {
                        "id": "OfficeRoom", 
                        "isPattern": "false", 
                        "type": "Room"
                    }
                ], 
                "providingApplication": "http://192.168.100.1:70/application"
            }
        }
    ]
}
+++++ Discovery 1 Done +++++++++++++++
+++++ Register 2: two attributes in vector +++++++++++++++
HTTP/1.1 200 OK
Content-Length: 75
Content-Type: application/json
Date: Mon, 02 Jun 2014 08:17:11 GMT

{
    "duration": "PT1M", 
    "registrationId": "538c3307c97e3502d1030c32"
}
+++++ Registration 2 Done +++++++++++++++
HTTP/1.1 200 OK
Content-Length: 599
Content-Type: application/json
Date: Mon, 02 Jun 2014 08:17:11 GMT

{
    "contextRegistrationResponses": [
        {
            "contextRegistration": {
                "attributes": [
                    {
                        "isDomain": "false", 
                        "name": "temperature", 
                        "type": "degree"
                    }, 
                    {
                        "isDomain": "false", 
                        "name": "pressure", 
                        "type": "bar"
                    }
                ], 
                "entities": [
                    {
                        "id": "BathRoom", 
                        "isPattern": "false", 
                        "type": "Room"
                    }
                ], 
                "providingApplication": "http://192.168.100.1:70/application"
            }
        }
    ]
}
+++++ Discovery 2 Done +++++++++++++++
